pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -787,3 +788,56 @@ def longest_common_subsequence(seq1: OneDimensionalArray, seq2: OneDimensionalAr
                     check_mat[i][j] = check_mat[i][j-1]
 
     return OneDimensionalArray(seq1._dtype, check_mat[row][col][-1])
+
+
+def longest_increasing_subsequence(array: Array) -> Integer:
+    """
+    Finds the length of longest increasing subsequence of the given sequences.
+
+    Parameters
+    ========
+
+    seq: OneDimensionalArray
+        The sequence.
+
+    Returns
+    =======
+
+    output: Integer
+        The length of longest increasing subsequence.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import longest_increasing_subsequence as LIS, OneDimensionalArray as ODA
+    >>> arr1 = ['1', '3, '9', '4', '6']
+    >>> lis = LIS(arr1)
+    >>> str(lis)
+    "4"
+    >>> arr1 = ['8', '7', '22', '12', '16', '28', '44']
+    >>> lcs = LIS(arr1)
+    >>> str(lis)
+    "[5"
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Longest_increasing_subsequence
+
+    Note
+   
+    """
+    n=len(seq)
+
+    lis=[1]*n
+
+    for i in range(1, n):
+        for j in range(0, i):
+            if seq[i] > seq[j] and seq[i]< seq[j] + 1 : 
+                lis[i] = lis[j]+1
```

Comment:
As far as I remember there is a binary search alternative as well for longest increasing subsequence. See, https://cp-algorithms.com/sequences/longest_increasing_subsequence.html

Could you please implement the above one? It will be more efficient at least asymptotically.